### PR TITLE
Fix language and market

### PR DESCRIPTION
**Pull request by StylianosGakis**
Before, sometimes on startup there'd be no language properly set and
we'd default back to en_SE, despite someone already having chosen a
different language, but in a way that was not understood by the
"migration" function.

Now we always run a "migration" function on startup, checking the
current market and setting the right language directly, or defaulting to
SE market if none was selected, and consequently selecting a language
in the process.

This should now work for all API levels in a consistent way, which will
not be cleared after the app is closed and started again.

### CODE REVIEW

File: app/feature/feature-login/src/main/kotlin/com/hedvig/android/feature/login/marketing/MarketingDestination.kt [hunk 1]
```diff
@@ -183,7 +185,7 @@ private fun MarketingScreen(
             modifier = Modifier.testTag("login_button"),
           )
           HedvigTextButton(
-            text = "Get a price quote",
```
Comment (hugokallstrom) [inline app/feature/feature-login/src/main/kotlin/com/hedvig/android/feature/login/marketing/MarketingDestination.kt:186]: Wops. Good find.

File: app/market/market-core/src/main/kotlin/com/hedvig/android/market/MarketStorage.kt [hunk 2]
```diff
@@ -11,14 +11,14 @@ import kotlinx.coroutines.flow.map
...
   private val datastore: DataStore<Preferences>,
 ) {
-  val market: Flow<Market> = datastore.data
+  val market: Flow<Market?> = datastore.data
     .map { preferences ->
       val storedMarket: Market? = preferences[marketPreferencesKey]?.let { Market.valueOf(it) }
       if (storedMarket != null) {
         return@map storedMarket
       } else {
         logcat { "No market stored in datastore, falling back to SE" }
```
Comment (hugokallstrom) [inline app/market/market-core/src/main/kotlin/com/hedvig/android/market/MarketStorage.kt:20]: Change this log message

**Review by hugokallstrom — APPROVED** [review]
Not trivial :D Good job!

File: app/feature/feature-login/src/main/kotlin/com/hedvig/android/feature/login/marketing/MarketingDestination.kt (hunk 1 above)
Comment (StylianosGakis) [reply app/feature/feature-login/src/main/kotlin/com/hedvig/android/feature/login/marketing/MarketingDestination.kt:186]: Yeah, I have no idea how this one slipped through, I see that I did some refactoring some months ago and I seem to have just never updated the localization, oops.

File: app/market/market-core/src/main/kotlin/com/hedvig/android/market/MarketStorage.kt (hunk 2 above)
Comment (StylianosGakis) [reply app/market/market-core/src/main/kotlin/com/hedvig/android/market/MarketStorage.kt:20]: Nice catch, thanks!

**Pull request merged**